packages/salesforcedx-vscode-apex/src/commands/forceLaunchReplayDebugger.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  EmptyParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from '@salesforce/salesforcedx-utils-vscode/out/src';
+import {
+  Command,
+  CommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import {
+  notificationService
+} from '@salesforce/salesforcedx-utils-vscode/out/src/commands';
+import * as vscode from 'vscode';
+import { nls } from '../messages';
+import {
+  testOutlineProvider
+} from '../views/testOutlineProvider';
+import {
+  forceAnonApexDebug
+} from './forceApexExecute';
+
+export async function forceLaunchReplayDebugger() {
+  const editor = vscode.window.activeTextEditor;
+  if (!editor) {
+    notificationService.showErrorMessage(
+      nls.localize('unable_to_locate_editor')
+    );
+    return;
+  }
+
+  const sourceUri = editor.document.uri;
+  if (!sourceUri) {
+    notificationService.showErrorMessage(
+      nls.localize('unable_to_locate_document')
+    );
+    return;
+  }
+
+  if (fileIsAnonymousApex(sourceUri)) {
+    await launchAnonymousApexReplayDebugger();
+    return;
+  }
+
+  const apexTestClassName = await getApexTestClassName(sourceUri);
+  if (apexTestClassName) {
+    await launchApexReplayDebugger(apexTestClassName);
+    return;
+  }
+
+  notificationService.showErrorMessage(
+    nls.localize('command_available_for_anon_apex_or_apex_test_only')
+  );
+}
+
+function fileIsAnonymousApex(sourceUri: vscode.Uri): boolean {
+  const extension = sourceUri.path.split('.').pop()?.toLowerCase();
+  return extension === 'apex';
+}
+
+async function getApexTestClassName(sourceUri: vscode.Uri): Promise<string | undefined> {
```

Comment:
should this go in an ApexService?

Reply:
Gotcha so this is talking to the view related class that holds the current list of test files.